zcash_test_vectors/orchard/key_components.py

```diff
@@ -54,7 +54,11 @@ def __init__(self, data):
 
 class FullViewingKey(object):
     def __init__(self, sk):
-        (self.rivk, self.ak, self.nk) = (sk.rivk, sk.ak, sk.nk)
+        if isinstance(sk, SpendingKey):
+            (self.rivk, self.ak, self.nk) = (sk.rivk, sk.ak, sk.nk)
+        else:
+            (self.rivk, self.ak, self.nk) = sk
```

Comment:
This would be clearer if we made the constructor take the three components (`(self, ak, nk, rivk)`), and then had two class methods for the different construction cases.

Reply:
We only need a single class method, `FullViewingKey.from_spending_key`, because the other case can use the constructor directly. Done.